crates/houston/src/error.rs

```diff
@@ -0,0 +1,31 @@
+use thiserror::Error;
+
+use std::io;
+
+/// HoustonProblem is the type of Error that occured.
+#[derive(Error, Debug)]
+pub enum HoustonProblem {
+    /// ConfigDirNotFound occurs when the default OS config can't be found.
+    #[error("Could not determine default OS config directory.")]
+    ConfigDirNotFound,
+
+    /// ProfileNotFound occurs when a profile with a specified name can't be found.
+    #[error("Profile {0} not found.")]
+    ProfileNotFound(String),
+
+    /// NoNonSensitiveConfigFound occurs when non-sensitive config can't be found for a profile.
+    #[error("No non-sensitive config found for profile {0}.")]
+    NoNonSensitiveConfigFound(String),
+
+    /// TomlSerialization occurs when a profile's configuration can't be serialized to a String.
+    #[error(transparent)]
+    TomlSerialization(#[from] toml::ser::Error),
```

Comment:
I'm a little suprised to see these implemented as `HoustonErrors`. Is that just to get the `from` casting so that we can use `?`? I'm a little unsure of how well this will scale?

Reply:
This is the way that `thiserror` recommends catching and returning errors and allows consumers of the library to know what types of error may arise within an application. This lets the crate keep a very clean API and does not require consumers of the library to import error types from the libraries that `Houston` uses under the hood. I'd recommend taking a look at [the docs for `thiserror`](https://docs.rs/thiserror/1.0.20/thiserror/)